src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <=0) {
+        Animal.alive.find();
+    }
```

Comment:
[CODE PERFORMANCE] The `find` method on line 30 is not properly used and will not remove the dead animal from the `Animal.alive` array. To follow the checklist, you should use `filter` instead to remove animals with health less than or equal to 0. Consider replacing the `find` method with a `filter` method on the `Animal.alive` array to ensure dead animals are removed.